core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -571,14 +573,16 @@
                               !xtql-idxer (delay (->xtql-indexer allocator live-idx-tx tx-ops-rdr ra-src wm-src scan-emitter tx-opts))
                               !sql-idxer (delay (->sql-indexer allocator live-idx-tx tx-ops-rdr ra-src wm-src scan-emitter tx-opts))]
                           (dotimes [tx-op-idx (.valueCount tx-ops-rdr)]
-                            (when-let [more-tx-ops (case (.getLeg tx-ops-rdr tx-op-idx)
-                                                     :xtql (.indexOp ^OpIndexer @!xtql-idxer tx-op-idx)
-                                                     :sql (.indexOp ^OpIndexer @!sql-idxer tx-op-idx)
-                                                     :put-docs (.indexOp ^OpIndexer @!put-docs-idxer tx-op-idx)
-                                                     :delete-docs (.indexOp ^OpIndexer @!delete-docs-idxer tx-op-idx)
-                                                     :erase-docs (.indexOp ^OpIndexer @!erase-docs-idxer tx-op-idx)
-                                                     :call (.indexOp ^OpIndexer @!call-idxer tx-op-idx)
-                                                     :abort (throw abort-exn))]
+                            (when-let [more-tx-ops
+                                       (.recordCallable ^Timer (:tx-timer metrics)
```

Comment:
(not one for now, your call)

When we write a few more of these, I might be tempted by a helper macro:

```clojure
(metrics/with-timer (:tx-timer metrics)
  (case (.getLeg ...)
    ...))
```